torch/utils/data/_utils/pin_memory.py

```diff
@@ -45,18 +45,30 @@ def _pin_memory_loop(in_queue, out_queue, device_id, done_event):
         del r  # save memory
 
 
-def pin_memory(data):
+def pin_memory(data, device=""):
     if isinstance(data, torch.Tensor):
-        return data.pin_memory()
+        if (len(device) == 0):
+            return data.pin_memory()
+        else:
+            return data.pin_memory(device)
     elif isinstance(data, string_classes):
         return data
     elif isinstance(data, collections.abc.Mapping):
-        return {k: pin_memory(sample) for k, sample in data.items()}
+        if (len(device) == 0):
+            return {k: pin_memory(sample) for k, sample in data.items()}
+        else:
+            return {k: pin_memory(sample, device) for k, sample in data.items()}
```

Comment:
@VitalyFedyunin is it preferable to check device==None here and omit from call to pin_memory?  or should we push device=None check into pin_memory and always pass device parameter (and simplify this code)

Reply:
You can always pass device=device in recurrent calls, without additional checks.

Reply:
will fix it.